blob/blob.go

```diff
@@ -144,16 +144,20 @@ func (b *Blob) Namespace() share.Namespace {
 }
 
 type jsonBlob struct {
-	Namespace    share.Namespace `json:"namespace"`
-	Data         []byte          `json:"data"`
-	ShareVersion uint32          `json:"share_version"`
-	Commitment   Commitment      `json:"commitment"`
+	Namespace share.Namespace `json:"namespace"`
+	// According to the Go documentation:
+	// Array and slice values encode as JSON arrays,
+	// except that []byte encodes as a base64-encoded string.
+	// So, this led to the incorrect representation of the `Data`.
+	Data         string     `json:"data"`
```

Comment:
I think it's better to keep []byte and make the client side aware of this instead. The reason it's base64 is data compression and we should not loose such property. As well as it will require changes for all the types returning []byte on the server.

Reply:
Client-side does not anything about any concrete type. It receives `json.RawMessage` as a result(where data is already marshalled incorrectly).

Reply:
It does not but the better solution to make it do. The RawMessage can be converted to Blob struct where base64 will converted automatically back.

Reply:
Ok

Reply:
After more time I realised, that it won't work as you described. the raw message is:
```
{
  "jsonrpc": "2.0",
  "result": [
    {
      "namespace": "AAAAAAAAAAAAAAAAAAAAAAAAAEJpDCBNOWAP3dM=",
      "data": "some_data"
      "share_version": 0,
      "commitment": "x0Z/D9P56ijvcVZ+s4al/d1cDNzQ+RaLNZKPUxITJNQ="
    }
  ],
  "id": 1
}
```

Raw message is not a Blob itself or jsonBlob. This is a struct and the `result` field is an empty interface with the underlying type of []jsobBlob. 
Also, it is clear that the "issue" is not on the unmarshaling part but on marshaling, just because golang marshals []byte this way. 
IMO, fixing it on the client side is not a good solution because we are fixing it only in our code, but anyone else who will be using blob API will face this issue and will have to deal with it(when will find the root cause 😄 ). 
My impression that keeping jsonBlob(not the Blob struct itself).Data as a string is the best solution. And yes, if we have other places w []byte, we need to fix it this way.

Reply:
n/a